### PR TITLE
Adding endpoints for seed's metering report configurations

### LINKED ISSUE #9183: Pepare API endpoints to interact with MeteringReportsConfiguration list
Currently we have a single endpoint `PUT /api/v1/admin/metering/configurations` responsible for modifying credentials. We need to deprecate it and introduce new API.

List all configurations:
`GET /api/v1/admin/metering/configurations/reports`

Get details of singe report configuration:
`GET /api/v1/admin/metering/configurations/reports/{report_configuration_name}`

Add new report configuration:
`POST /api/v1/admin/metering/configurations/reports/{report_configuration_name}`
```json
// payload
{
  "name": "daily",
  "interval": 1,
  "schedule": "0 1 * * *"
}
```

Update report configuration:
`PUT /api/v1/admin/metering/configurations/reports/{report_configuration_name}`
```json
// payload
{
  "interval": 1,
  "schedule": "0 1 * * *"
}
Delete new report configuration:
`DELETE /api/v1/admin/metering/configurations/reports/{report_configuration_name}`

### PR DESCRIPTION
**Pull request by pkprzekwas**
**What does this PR do / Why do we need it**:
Created endpoints to interact with Seed's metering report configurations.

**Does this PR close any issues?**:
Fixes #9183 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/ee/metering/report_config_handler.go [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+	// in: path
+	// required: true
+	ReportConfigurationName string `json:"report_configuration_name"`
+}
+
+// swagger:parameters createMeteringReportConfiguration
+type createReportConfigurationReq struct {
+	// in: body
+	Body struct {
+		Name     string `json:"name"`
```
Comment (zreigz) [inline pkg/ee/metering/report_config_handler.go]: I guess this name is the same as `ReportConfigurationName`. I would suggest making it simple and using just `Name` everywhere. 

In other requests, you have the `Name` out of the body. Should be the same here.

**Review by zreigz — CHANGES_REQUESTED** [review]

File: pkg/ee/metering/report_config_handler.go [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+	}
+
+	return nil
+}
+
+// swagger:parameters updateMeteringReportConfiguration
+type updateReportConfigurationReq struct {
+	// in: path
+	// required: true
+	Name string `json:"report_configuration_name"`
```
Comment (zreigz) [inline pkg/ee/metering/report_config_handler.go]: please change json for `name`

File: pkg/ee/metering/report_config_handler.go [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+	"k8c.io/kubermatic/v2/pkg/validation"
+
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// swagger:parameters getMeteringReportConfiguration
+type getMeteringReportConfig struct {
+	// in: path
+	// required: true
+	Name string `json:"report_configuration_name"`
```
Comment (zreigz) [inline pkg/ee/metering/report_config_handler.go]: please change json for name

File: pkg/ee/metering/report_config_handler.go [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+}
+
+// swagger:parameters deleteMeteringReportConfiguration
+type deleteMeteringReportConfig struct {
+	// in: path
+	// required: true
+	Name string `json:"report_configuration_name"`
```
Comment (zreigz) [inline pkg/ee/metering/report_config_handler.go]: please change json for name

File: pkg/ee/metering/report_config_handler.go (hunk 1 above)
Comment (zreigz) [reply pkg/ee/metering/report_config_handler.go]: can you move the name out of the body

**Review by zreigz — CHANGES_REQUESTED** [review]

File: pkg/ee/metering/report_config_handler.go (hunk 2 above)
Comment (pkprzekwas) [reply pkg/ee/metering/report_config_handler.go]: Oh, I though you were talking about the variable naming. My bad, will update the json.

File: pkg/ee/metering/report_config_handler.go (hunk 1 above)
Comment (pkprzekwas) [reply pkg/ee/metering/report_config_handler.go]: Oh, so you suggest keeping the `Name` as the path param?  Actually, now that I think about it, it makes sense.

**Pull request merged**